### PR TITLE
Fix extra checks

### PR DESCRIPTION
**Pull request by BramMeir**
* Option to show the generated artifacts to the students
* Option to remove private docker images
* Fix new students join individual projects automatically
* Remove emojis after democratic decision

### CODE REVIEW

File: backend/api/serializers/course_serializer.py [hunk 1]
```diff
@@ -73,6 +90,7 @@ class Meta:
             "excerpt",
             "description",
             "faculty",
+            "faculty_id",
```
Comment (francisvaut) [inline backend/api/serializers/course_serializer.py:93]: Why does the course serializer now contain both the faculty and the faculty_id apart?

File: frontend/src/assets/lang/app/nl.json [hunk 2]
```diff
@@ -44,21 +44,21 @@
         },
         "calendar": {
             "title": "Kalender",
-            "noProjects": "Geen projecten op geselecteerde datum. \uD83E\uDD73"
```
Comment (francisvaut) [inline frontend/src/assets/lang/app/nl.json:47]: :disappointed:

File: frontend/src/composables/services/course.service.ts
Comment (francisvaut) [inline frontend/src/composables/services/course.service.ts:1]: Why these changes?

**Review by francisvaut — CHANGES_REQUESTED** [review]
Looks good overall, but I don't know what exactly goes wrong but when creating a project with multiple extra checks and different combinations for artifacts and logs visibilty, I get the following error:
```
ValueError: Field 'id' expected a number but got ''.
"POST /api/projects/4/extra_checks/ HTTP/1.0" 500 19216
 Internal Server Error: /api/projects/4/extra_checks/
 ```

File: backend/api/serializers/course_serializer.py (hunk 1 above)
Comment (BramMeir) [reply backend/api/serializers/course_serializer.py:93]: These changes were to remove the CreateCourseSerializer and put all the logic in the CourseSerializer to make it cleaner. But by using the FacultySerializer and passing a faculty object in a POST request to create a course, this fails when there is already a faculty with that ID ( but that is always the case because you never make a faculty, you just pass it to link it to the course).
Using this faculty_id field, you can pass a faculty_id in a request and this will automatically link the corresponding faculty to the course ( btw this field is not displayed in a GET request, only used in creation).

File: frontend/src/composables/services/course.service.ts
Comment (BramMeir) [reply frontend/src/composables/services/course.service.ts:1]: See other comment

**Comment by BramMeir** [comment]
> Looks good overall, but I don't know what exactly goes wrong but when creating a project with multiple extra checks and different combinations for artifacts and logs visibilty, I get the following error:
> 
> ```
> ValueError: Field 'id' expected a number but got ''.
> "POST /api/projects/4/extra_checks/ HTTP/1.0" 500 19216
>  Internal Server Error: /api/projects/4/extra_checks/
> ```

I see the issue, when you create multiple extra checks the form is reset. But by doing this, the docker image field is set to a default empty one. So when you don't select a docker image in the list, this passes the validation but will not work when making the request. I will fix that.

**Review by francisvaut — APPROVED** [review]
Nice :raised_hands:
